faithcomesbyhearing/dbp-web
# Ticket-2238 Update the Bible.is web dependencies to use the latest ve…

# Description
Update the Bible.is web dependencies to use the latest version of each one. The main changes are:

1. Migrated to SWC using Next.js.
2. Migrated Jest test files to use @testing-library/react instead of Enzyme.
3. Removed Babel dependencies for compilation.
4. Upgraded ESLint to version 9.
5. Removed unnecessary dependencies such as Webpack, as the latest Next.js version already incorporates this.
6. Supported Node.js version 20.12.2, which is the version supported by AWS (see: https://docs.aws.amazon.com/elasticbeanstalk/latest/platforms/platform-history-nodejs.html).

# Task
[User Story 2238](https://dev.azure.com/keyholesoftware/FCBH%20-%20Bible%20Brain/_workitems/edit/2238): bibleis web: Update nodejs to version 20

# How to test
Run Jest tests and ensure all tests pass.

```shell
npm run test
```
- Outcome
```shell
> dbp-reader@3.5.0 test
> env NODE_ENV=test jest -u

 PASS  app/components/Checkbox/tests/index.test.js
 PASS  app/containers/ChapterSelection/tests/index.test.js
 PASS  app/components/BooksTestament/tests/index.test.js
 PASS  app/containers/Text/tests/index.test.js
 PASS  app/components/NavigationBar/tests/index.test.js
 PASS  app/components/Information/tests/index.test.js
 PASS  app/containers/Text/tests/highlightFormatted.test.js
 PASS  app/components/HighlightColors/tests/index.test.js
 PASS  app/utils/tests/getBookMetaData.test.js
 PASS  app/utils/tests/highlightingUtils.test.js
 PASS  app/components/FootnotePortal/tests/index.test.js
 PASS  app/components/VersionListSection/tests/index.test.js
 › 2 snapshots updated.
 PASS  app/components/AudioDramaToggle/tests/index.test.js
 PASS  app/utils/tests/checkForVideo.test.js
 PASS  app/components/PlainTextVerse/tests/index.test.js
 PASS  app/components/JesusFilmVideoPlayer/tests/index.test.js
 PASS  app/components/AudioProgressBar/tests/index.test.js
 PASS  app/components/FormattedText/tests/index.test.js
 PASS  app/utils/tests/getInitialChapterData.test.js
 PASS  app/components/ContextPortal/tests/index.test.js
 PASS  app/components/Chapter/tests/index.test.js
 PASS  app/components/PlainText/tests/index.test.js
 PASS  app/components/LanguageList/tests/index.test.js
 PASS  app/components/CopyrightStatement/tests/index.test.js
 PASS  app/utils/tests/cachecFetch.test.js
 PASS  app/components/DonateButton/tests/index.test.js
 PASS  app/components/EditNote/tests/index.test.js
 PASS  app/components/VersionList/tests/index.test.js
 › 1 snapshot updated.
 PASS  app/utils/tests/setEventHandlersForFootnotes.test.js
 PASS  app/components/AudioOnlyMessage/tests/index.test.js
 PASS  app/components/ReadFullChapter/tests/index.test.js
 PASS  app/components/NewChapterArrow/tests/index.test.js
 PASS  app/components/ReadersModeVerse/tests/index.test.js
 PASS  app/utils/tests/getAudioAsyncCall.test.js
 PASS  app/components/ColorPicker/tests/index.test.js
 PASS  app/components/MyNotes/tests/index.test.js
 PASS  app/utils/tests/hreLinkOrAsLink.test.js
 PASS  app/utils/tests/setEventHandlersForFormattedVerses.test.js
 PASS  app/components/MyHighlights/tests/index.test.js
 PASS  app/components/Footer/tests/index.test.js
 PASS  app/components/PasswordReset/tests/index.test.js
 PASS  app/components/CountryList/tests/index.test.js
 PASS  app/components/MyBookmarks/tests/index.test.js
 PASS  app/utils/tests/parseCookie.test.js
 PASS  app/utils/tests/deepDifferenceObject.test.js
 PASS  app/utils/tests/bugsnagClient.test.js
 PASS  app/utils/tests/request.test.js
 PASS  app/utils/tests/getFirstChapterReference.test.js
 PASS  app/utils/tests/bugsnagServer.test.js
 PASS  app/components/PlaybackRateSvg/tests/index.test.js
 PASS  app/utils/tests/reconcilePersistedState.test.js
 PASS  app/utils/tests/checkStore.test.js
 PASS  app/utils/tests/getNextChapterUrl.test.js
 PASS  app/components/AccountSettings/tests/index.test.js
 PASS  app/utils/tests/getClassNamesForAudio.test.js
 PASS  app/components/CopyrightSection/tests/index.test.js
 PASS  app/utils/tests/isUserAgentInternetExplorer.test.js
 PASS  app/containers/Text/tests/highlightPlain.test.js
 PASS  app/components/NoteVerseText/tests/index.test.js
 PASS  app/components/Login/tests/index.test.js
 PASS  app/components/FadeTransition/tests/index.test.js
 PASS  app/utils/tests/checkBookName.test.js
 PASS  app/components/PleaseSignInMessage/tests/index.test.js
 PASS  app/tests/i18n.test.js
 PASS  app/utils/tests/getFirstMatchingChapter.test.js
 PASS  app/utils/tests/checkEmailForValidity.test.js
 PASS  app/utils/tests/stateReconciler.test.js
 PASS  app/utils/tests/getPreviousChapterUrl.test.js
 PASS  app/utils/tests/closeMenuFunctions.test.js
 PASS  app/utils/tests/reduxPersist.test.js
 PASS  app/utils/tests/listOfBooksInBible.test.js
 PASS  app/utils/tests/get.test.js
 PASS  app/utils/tests/getFormattedTimeString.test.js
 PASS  app/utils/tests/constants.test.js
 PASS  app/components/BooksTable/tests/index.test.js (6.855 s)

Snapshot Summary
 › 3 snapshots updated from 2 test suites.

Test Suites: 75 passed, 75 total
Tests:       2 skipped, 411 passed, 413 total
Snapshots:   3 updated, 101 passed, 104 total
Time:        7.75 s
Ran all test suites.

```
